app/scripts/contentscript.js

```diff
@@ -103,6 +104,23 @@ const IGNORE_INIT_METHODS_FOR_KEEP_ALIVE = [
 let keepAliveInterval;
 let keepAliveTimer;
 
+/**
+ * Sending a message to the extension to receive will keep the service worker alive.
+ *
+ * If the extension is unloaded or reloaded during a session and the user attempts to send a
+ * message to the extension, an "Extension context invalidated." error will occur.
+ * When this happens, prompt the user to reload the extension.
+ */
+const sendMessageWorkerKeepAlive = () => {
+  browser.runtime
+    .sendMessage({ name: WORKER_KEEP_ALIVE_MESSAGE })
+    .then(null, (error) => {
+      error.message === 'Extension context invalidated.'
+        ? log.error(`Please refresh the page. MetaMask: ${error}`)
+        : log.error(`MetaMask: ${error}`);
+    });
+};
```

Comment:
in case it helps, see how to handle `browser.runtime.sendMessage` errors here: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/sendMessage#examples